fuel-compression/src/lib.rs

```diff
@@ -0,0 +1,19 @@
+//! Compression and decompression of fuel-types for the DA layer
+
+#![cfg_attr(docsrs, feature(doc_auto_cfg))]
+#![warn(missing_docs)]
+#![deny(unsafe_code)]
+#![deny(unused_crate_dependencies)]
+#![deny(clippy::cast_possible_truncation)]
+
+mod impls;
+mod key;
+mod traits;
+
+pub use key::RegistryKey;
+pub use traits::*;
+
+pub use fuel_derive::{
+    Compress,
+    Decompress,
+};
```

Comment:
What do you think about renaming macros into `CompressibleBy` and `DecompressibleBy` to show which traits are implemented?

Reply:
I see the point in having the same name here. I dislike it since the names are confusing. When I see

```rust
#[derive(fuel_compression::Compress)]
```

I know this type can be compressed by some `fuel_compression`-related machinery. When I see

```rust
#[derive(fuel_compression::CompressibleBy)]
```

I'm like compressible by what? `Compressible` might be an improvement.

Also, `Compress` implements both `Compressible` and `CompressibleBy`, and the define macros having the same names as the trait obfuscates this fact even more.

Reply:
> Also, Compress implements both Compressible and CompressibleBy, and the define macros having the same names as the trait obfuscates this fact even more.

Yeah, it is true. But `Compressible` is a super trait of `CompressibleBy`, so it is not so bad=D But I agree that it is hidden property of the macro that can be discovered only from the comment.

But in the case of `DecompressibleBy` everything is clear=) The macro implements the same trait.

But it is up to you, I will not push it futher=)